File: PythonServer/FinalPythonShop/src/OrderLine.py
import Item
class OrderLine:
    #private Item item;
    #private int quantity;

    def __init__(self, item:Item, quantity:int):
        self.item = item
        self.quantity = quantity

    def subTotal(self):
        return self.item.getPrice() * self.quantity

    def toString(self):
        return str("Line:{%5d x %s = $%10.2f}",self.getQuantity(),self.getItem(), self.subTotal())

    def getItem(self):
        return self.item

    def setItem(self, item:Item):
        self.item = item

    def getQuantity(self):
        return self.quantity

    def setQuantity(self, quantity:int):
        self.quantity = quantity

File: ClientDisplay/src/catalog/OrderLine.java
package catalog;

public class OrderLine {
    private Item item;
    private int quantity;

    public OrderLine(Item item, int quantity) {
        this.item = item;
        this.quantity = quantity;
    }

    public double subTotal(){
        return this.item.getPrice() * this.quantity;
    }

    public String toString(){
        return String.format("Line:{%5d x %s = $%10.2f}",this.getQuantity(),this.getItem(), this.subTotal());
    }

    public Item getItem() {
        return item;
    }

    public void setItem(Item item) {
        this.item = item;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }
}